src/west/app/project.py

```diff
@@ -845,6 +855,77 @@ def handle(group_filter_item):
 
         self.fs = self.fetch_strategy()
 
+    @staticmethod
+    def _update_concurrent_thread(args):
+        project, self, logdir = args
+        if logdir:
+            stdout = os.dup(sys.stdout.fileno())
+            stderr = os.dup(sys.stderr.fileno())
+
+            logfile = open(os.path.join(logdir, project.name), "wb")
+            os.dup2(logfile.fileno(), sys.stdout.fileno())
+            os.dup2(logfile.fileno(), sys.stderr.fileno())
+
+        try:
```

Comment:
I would remove the outer try, I don't think it has any purpose here. A single try / except / finally should do the same thing.